packages/dds/tree/src/simple-tree/api/schemaCompatibilityTester.ts

```diff
@@ -93,10 +95,16 @@ export class SchemaCompatibilityTester {
 					if (result === PosetComparisonResult.Greater) {
 						// Stored schema is more relaxed than view schema.
 						canUpgrade = false;
+						// Get the node this field is under, if any:
+						const treeNodeSchema =
+							discrepancy.identifier === undefined
+								? undefined
+								: this.viewSchema.definitions.get(discrepancy.identifier);
 						if (
 							discrepancy.view === FieldKinds.forbidden.identifier &&
-							discrepancy.identifier !== undefined &&
-							this.policy.allowUnknownOptionalFields(discrepancy.identifier)
```

Comment:
We can get this same information from the view schema, and it makes way more sense to do so (as done in this PR). This change also is removing policy.allowUnknownOptionalFields since not of the use of it really makes sense.